using-mobile-extensions/adobe-journey-optimizer-decisioning/api-reference.md

```diff
@@ -0,0 +1,542 @@
+
+```swift
+/// `Offer` class
+@objc(AEPOffer)
+public class Offer: NSObject, Codable {
+    /// Unique Offer identifier
+    @objc public let id: String
+
+    /// Offer revision detail at the time of the request
+    @objc public let etag: String
+
+    /// Offer schema string
+    @objc public let schema: String
+
+    /// Offer type as represented in enum `OfferType`
+    @objc public let type: OfferType
+
+    /// Optional Offer language array
+    @objc public let language: [String]?
+
+    /// Offer content string
+    @objc public let content: String
+
+    /// Optional Offer characteristics dictionary
+    @objc public let characteristics: [String: String]?
+
+    /// Weak reference to Proposition instance
+    @objc weak var proposition: Proposition?
+}
+```
+
+The `Offer` class extension provides methods for generating XDM data for Proposition Interactions field group which can be used for proposition tracking. It also contains direct methods for  tracking proposition display and tap interactions.
+
+```swift
+/// `Offer` extension
+public extension Offer {
+    /// Creates a dictionary containing XDM formatted data for `Experience Event - Proposition Interactions` field group from the given proposition option.
+    ///
+    /// The Edge `sendEvent(experienceEvent:_:)` API can be used to dispatch this data in an Experience Event along with any additional XDM, free-form data, or override dataset identifier.
+    ///
+    /// - Note: The returned XDM data also contains the `eventType` for the Experience Event with value `display`.
+    /// - Returns A dictionary containing XDM data for the propositon interactions.
+    func generateDisplayInteractionXdm() -> [String: Any] {...}
+
+    /// Creates a dictionary containing XDM formatted data for `Experience Event - Proposition Interactions` field group from the given proposition option.
+    ///
+    /// The Edge `sendEvent(experienceEvent:_:)` API can be used to dispatch this data in an Experience Event along with any additional XDM, free-form data, or override dataset identifier.
+    ///
+    /// - Note: The returned XDM data also contains the `eventType` for the Experience Event with value `click`.
+    /// - Returns A dictionary containing XDM data for the propositon interactions.
+    func generateTapInteractionXdm() -> [String: Any] {...}
+
+    /// Dispatches an event for the Edge extension to send an Experience Event to the Edge network with the display interaction data for the given proposition item.
+    func displayed() {...}
+
+    /// Dispatches an event for the Edge extension to send an Experience Event to the Edge network with the tap interaction data for the given proposition item.
+    func tapped() {...}
+}
+```
+
+### OfferType
+
+An enum indicating the type of an Offer, derived from the proposition item `format` field in personalization query response.
```

Comment:
I don't think offer needs to be capitalized here. So it can just be "... the type of an offer, derived from..."